packages/react-data-grid/src/utils/viewportUtils.ts

```diff
@@ -63,12 +63,12 @@ export function getHorizontalRangeToRender<R>({
   let columnIndex = lastFrozenColumnIndex;
   let hiddenColumnsWidth = 0;
   while (remainingScroll >= 0 && columnIndex < columns.length) {
```

Comment:
I believe we only need to do this.
```suggestion
  while (remainingScroll >= 0 && columnIndex <= columns.length - 2) {
```

Reply:
why skipping index 0

Reply:
hmm... I think there is another bug, if select column is enabled, i.e. the checkbox column, then we should skip the 0, I believe we need to check exactly like the start condition. However, the end condition should be correct.

wdyt? @MayhemYDG

Reply:
@qili26 No? The select column isn't special in this context.

Reply:
nvm, I believe I mixed his change and the original code when leaving the comment. If there is no frozen columns, then the `lastFrozenColumnIndex` would just return `-1`, right? So his change for this `Math.max(lastFrozenColumnIndex, 0)` is not needed.

For the loop break condition, shouldn't we do `columnIndex <= columns.length - 2`? The current code will exceed the array boundary and get an `undefined` column, which will break `column.width` when we get the width.

let's say `columns` has 10 elements, and `columnIndex` is 9, `columnIndex++` will result in 10 and `columns[10]` is `undefined` and then we do `remainingScroll -= undefined.width;`. Am I missing anything here?

Reply:
>If there is no frozen columns, then the lastFrozenColumnIndex would just return -1, right? So his change for this Math.max(lastFrozenColumnIndex, 0) is not needed.

Correct.

>For the loop break condition, shouldn't we do `columnIndex <= columns.length - 2`?

Possibly, I'm not 100% certain. Addressing the "impossible" scroll left value fixes this edge case,  and maybe more, so I think it's enough.

>Am I missing anything here?

As long as we use valid dimensions/scroll positions, then `remainingScroll >= 0` will break us out of the loop before we try to get an out of bounds column.

Reply:
> Possibly, I'm not 100% certain. Addressing the "impossible" scroll left value fixes this edge case, and maybe more, so I think it's enough.
so `columnIndex < columns.length` will always return true then, why keeping it

Reply:
>so `columnIndex < columns.length` will always return true then, why keeping it

You're right, let's remove it.

This method is kind of confusing, I think I'll refactor it after this PR.